src/app/auth.service.ts

```diff
@@ -0,0 +1,48 @@
+import { UserService } from './user.service';
+import { User } from './models/user';
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/auth';
+import * as firebase from 'firebase';
+import { ActivatedRoute } from '@angular/router';
+import { switchMap } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  userObservable: Observable<firebase.User>;
```

Comment:
private?

Reply:
I'm using this in other components:  admin-auth-guard, auth-guard, and app